### PR TITLE
fix(app-degree-pages): update major map filtering to include all curr…

### PR DESCRIPTION
**Pull request by davidornelas11**
…ent year maps

This also adds a test which uses the nursing degree as mock data to make sure the 4 major maps show up correctly. This seems to cover the edge case

## Checklist

- [x] Tests pass for relevant code changes

### Links

- [JIRA ticket](https://asudev.jira.com/browse/UDS-2094)

### CODE REVIEW

File: packages/app-degree-pages/src/core/services/degree-data-prop-resolver-service.jsx [hunk 1]
```diff
@@ -27,6 +27,58 @@ const getMajorityOwner = row => {
...
+ * This service acts as a data accessor layer for degree program information, providing a clean API
+ * to retrieve various properties and computed values from raw degree data.
+ *
+ * @param {Object} [row={}] - The raw degree data object containing all degree program information
+ * @param {string} [row.acadPlanMarketingDescription] - Marketing description of the academic plan
+ * @param {string} [row.acadPlanCode] - Academic plan code
+ * @param {string} [row.degreeDescriptionShort] - Short description of the degree
+ * @param {string} [row.acadPlanTypeDescription] - Type description of the academic plan
+ * @param {Array<Object>} [row.majorMapGeneral] - Array of general major maps
+ * @param {string} [row.degreeType] - Type of degree (e.g., "GR" for graduate, "UGCM" for undergraduate)
```
Comment (mlsamuelson) [inline packages/app-degree-pages/src/core/services/degree-data-prop-resolver-service.jsx:41]: Really like the inclusion of these details, especially the descriptive per-field/param details.

**Review by mlsamuelson — APPROVED** [review]
